src/plugins/intel_npu/src/compiler/include/custom_stream_buffer.hpp

```diff
@@ -0,0 +1,128 @@
+    }
+
+    virtual std::streamsize xsputn(const char* s, std::streamsize n) override {
+        if (_writePos + n > _buffer.size()) {
+            _buffer.resize(_writePos + n);
+        }
+        std::copy(s, s + n, _buffer.data() + _writePos);
+        _writePos += n;
+        this->setp(_buffer.data(), _buffer.data() + _buffer.size());
+        this->pbump(static_cast<int>(_writePos));
+        return n;
+    }
+
+    virtual std::streamsize xsgetn(char* s, std::streamsize n) override {
+        std::streamsize num = std::min(n, static_cast<std::streamsize>(_buffer.size() - _readPos));
+        std::copy(_buffer.data() + _readPos, _buffer.data() + _readPos + num, s);
+        _readPos += num;
+        this->setg(_buffer.data(), _buffer.data() + _readPos, _buffer.data() + _buffer.size());
+        return num;
+    }
+
+    virtual std::streampos seekoff(std::streamoff off,
+                                   std::ios_base::seekdir way,
+                                   std::ios_base::openmode which = std::ios_base::in | std::ios_base::out) override {
+        if (which & std::ios_base::in) {
+            switch (way) {
+            case std::ios_base::beg:
+                _readPos = off;
+                break;
+            case std::ios_base::cur:
+                _readPos += off;
+                break;
+            case std::ios_base::end:
+                _readPos = _buffer.size() + off;
+            default:
+                break;
+            }
+
+            if (_readPos > _buffer.size() || _readPos < 0) {
+                return std::streampos(std::streamoff(-1));
+            }
+            this->setg(_buffer.data(), _buffer.data() + _readPos, _buffer.data() + _buffer.size());
+            return std::streampos(_readPos);
+        }
+
+        if (which & std::ios_base::out) {
+            switch (way) {
+            case std::ios_base::beg:
+                _writePos = off;
+                break;
+            case std::ios_base::cur:
+                _writePos += off;
+                break;
+            case std::ios_base::end:
+                _writePos = _buffer.size() + off;
+            default:
+                break;
+            }
+
+            if (_writePos > _buffer.size() || _writePos < 0) {
+                return std::streampos(std::streamoff(-1));
+            }
+            this->setp(_buffer.data(), _buffer.data() + _buffer.size());
+            this->pbump(static_cast<int>(_writePos));
+            return std::streampos(_writePos);
+        }
+        return std::streampos(std::streamoff(-1));
+    }
+
+    virtual std::streampos seekpos(std::streampos pos,
+                                   std::ios_base::openmode which = std::ios_base::in | std::ios_base::out) override {
+        return seekoff(static_cast<std::streamoff>(pos), std::ios_base::beg, which);
+    }
+
+private:
+    // TODO: Use deque as a buffer to avoid reallocation
+    std::vector<char> _buffer;
+    size_t _readPos;
```

Comment:
In this implementation the positions are equivalent to the gptr/pptr pointers from the base class.